scrapegraphai/nodes/merge_answers_node.py

```diff
@@ -65,31 +63,43 @@ def execute(self, state: dict) -> dict:
         user_prompt = input_data[0]
         answers = input_data[1]
 
-        # merge the answers in one string
-        answers_str = ""
-        for i, answer in enumerate(answers):
-            answers_str += f"CONTENT WEBSITE {i+1}: {answer}\n"
-
-        # Initialize the output parser
-        if self.node_config.get("schema", None) is not None:
-            output_parser = JsonOutputParser(pydantic_object=self.node_config["schema"])
+        #skip merge answers if there are no multiple answers to merge
+        if len(answers) > 1:
```

Comment:
you refactored the node because you expect search nodes might return single answers, hence the change; in that case I would prefer the use of a conditional node (which we have the class for, but not yet implemented so far) and keep the merge answers node as it is for multiple answers; sounds good to me if you could implement it and refactor it

Reply:
When calling merge_answer_node in a recursive manner inside the deepscrapegraph, leaf nodes in the URL tree usually contain 1-2 links, specifically after filtering the URLs. Don't you think it would be better to cover all the cases inside the merge answer?

Reply:
I think you've got a point, but I'm worried that would lead to further if/else branching in the future if new edge cases come up that we have to carefully handle inside any given node's body

In the general case, that feels like a conditional that should be applied before the node execution